diff --git a/node_modules/metro/src/Assets.js b/node_modules/metro/src/Assets.js
index ac3302c..1c0c49a 100644
--- a/node_modules/metro/src/Assets.js
+++ b/node_modules/metro/src/Assets.js
@@ -222,7 +222,15 @@ async function getAsset(
     );
   }
   if (fileExistsInFileMap != null) {
-    if (!fileExistsInFileMap(absolutePath)) {
+    let fileFound = fileExistsInFileMap(absolutePath);
+    if (!fileFound && platform != null) {
+      const platformPath = absolutePath.replace(
+        /\.([^.]+)$/,
+        `.${platform}.$1`,
+      );
+      fileFound = fileExistsInFileMap(platformPath);
+    }
+    if (!fileFound) {
       throw new Error(
         `'${relativePath}' could not be found, because it is not within the projectRoot or watchFolders, or it is blocked via the resolver.blockList config`,
       );
